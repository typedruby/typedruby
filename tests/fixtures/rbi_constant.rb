# @typedruby
#
# Demonstrate that constants can be typed via .rbi files
class A
  STRS = ['a', 'b'].freeze

  def use_strs
    STRS
  end
end
