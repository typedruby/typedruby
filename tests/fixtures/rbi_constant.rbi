class A
  STRS = (nil : Array::[String])

  def use_strs => Array::[Symbol]; end
end
