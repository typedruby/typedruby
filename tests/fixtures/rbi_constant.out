error: Could not match types:

        @ __ROOT__/tests/fixtures/rbi_constant.rbi:4
      4 |    def use_strs => Array::[Symbol]; end
                                     ^^^^^^ Symbol, with:
        @ __ROOT__/tests/fixtures/rbi_constant.rbi:2
      2 |    STRS = (nil : Array::[String])
                                   ^^^^^^ String

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/rbi_constant.rbi:4
      4 |    def use_strs => Array::[Symbol]; end
                             ^^^^^^^^^^^^^^^ Array::[Symbol], with:
        @ __ROOT__/tests/fixtures/rbi_constant.rbi:2
      2 |    STRS = (nil : Array::[String])
                           ^^^^^^^^^^^^^^^ Array::[String]
